interfaces/repo_test.go

```diff
@@ -1239,3 +1239,57 @@ func (s *RepositorySuite) TestAutoConnectContentInterfaceNoMatchingDeveloper(c *
 	candidateSlots := repo.AutoConnectCandidates("content-plug-snap", "import-content")
 	c.Check(candidateSlots, HasLen, 0)
 }
+
+func makeLivepatchConnectionTestSnaps(c *C, name, developer string) (*Repository, *snap.Info, *snap.Info) {
+	repo := NewRepository()
+	err := repo.AddInterface(&TestInterface{InterfaceName: "restricted", AutoConnectFlag: false})
+
+	plugSnap, err := snap.InfoFromSnapYaml([]byte(fmt.Sprintf(`
+name: %s
+plugs:
+  canonical-livepatch:
+    interface: restricted
+`, name)))
+	c.Assert(err, IsNil)
+	slotSnap, err := snap.InfoFromSnapYaml([]byte(`
+name: ubuntu-core
+type: os
+slots:
+  restricted:
+    interface: restricted
+`))
+	c.Assert(err, IsNil)
+
+	plugSnap.DeveloperID = developer
+	slotSnap.DeveloperID = "canonical"
+	slotSnap.Type = snap.TypeOS
+
+	err = repo.AddSnap(plugSnap)
+	c.Assert(err, IsNil)
+	err = repo.AddSnap(slotSnap)
+	c.Assert(err, IsNil)
+
+	return repo, plugSnap, slotSnap
+}
```

Comment:
I'm surprised this is all it took to declare a new interface in the os snap....

Can you add a couple more tests:
- where isSpecialAutoConnectSnap() == true and an autoconnectable interface (eg, network) and a non-autoconnectable interface (eg, kernel-module-control) both autoconnect
- where isSpecialAutoConnectSnap() == false and an autoconnectable interface (eg, network) connects but a non-autoconnectable interface (eg, kernel-module-control) does not?